front-end/src/components/analysis-page/AnalysisPage.jsx

```diff
@@ -3,38 +3,78 @@ import { connect } from "react-redux";
 import store from "../../store.js";
 import { openMainPage } from "../../store/switchPageSlice.js";
 import Logo from "../icons/logo.jsx";
+import BaseButton from "../general/BaseButton.jsx"; // REMOVE!
 import IntentLister from "./IntentLister.jsx";
+import IntentMenu from "./IntentMenu.jsx";
 
 export class AnalysisPage extends Component {
 
+    state = {
+      showIntentMenu: false,
+    }
+
+    toggleIntentMenu = () => {
+      this.setState({showIntentMenu: !this.state.showIntentMenu});
+    }
+  
+    hideIntentMenu = () => {
+      // Hide intent menu; use instead of toggleIntentMenu in scenarios when hiding the menu is the only desired outcome
+      this.setState({showIntentMenu: false});
+    }
+  
     openMainPage = () => {
       store.dispatch(openMainPage());
     }
 
     render() {
       let projectId = this.props.projectId;
       let analyzedTranscripts = this.props.analyzedTranscripts;
-        return (
-            <div className="AnalysisPage bg-purple-100 flex h-screen" data-testid="analysis-page">
-              <div className="flex gap-y-10 w-full flex-col h-full">
-                <div /*second line of className is tooltip styling*/
-                  className="justify-left relative text-center text-sm
-                    before:z-10 before:absolute before:-right-5 before:top-1/2 before:w-max before:max-w-xs before:translate-x-full before:-translate-y-1/2 before:rounded-md before:bg-gray before:px-3 before:py-2 before:text-off-white before:invisible before:content-[attr(tooltip)] after:z-10 after:absolute after:-right-[0.8rem] after:top-1/2 after:h-0 after:w-0 after:translate-x-2 after:-translate-y-1/2 after:border-8 after:border-r-gray after:border-l-transparent after:border-b-transparent after:border-t-transparent after:invisible hover:before:visible hover:after:visible
-                    cursor-pointer m-5 w-[5em]"
-                  role="button"
-                  onClick={this.openMainPage}
-                  aria-label="Home Button"
-                  tooltip="Click here to return to homepage"
-                  data-testid="logo-button"
-                >
-                  <Logo />
-                </div>
-                <div className="w-4/5 h-4/5 mx-auto">
-                  <IntentLister intents={analyzedTranscripts[projectId]} />
-                </div>
+      return (
+        <div className="AnalysisPage" data-testid="analysis-page">
+          <div /* display sidebar if toggled on */>
+            <IntentMenu
+              intents={analyzedTranscripts[projectId]}
+              isOpen={this.state.showIntentMenu}
+              onClickOutside={this.hideIntentMenu}
+            />
+          </div>
+          <div /* darken screen if sidebar is toggled on */
+            className={"bg-purple-100 h-screen transition ease-in-out" + (this.state.showIntentMenu ? " brightness-75" : "")}
+            data-testid="analysis-page-main"
+          >
```

Comment:
Brilliant! The one thing is, if the brightness is lowered, check if we should make a11y changes (i.e. if there is an aria attribute we should be applying)

Reply:
Sure - I want to clarify what this means. Is there one attribute to apply to the whole div or specific attributes to each of the elements in the div? I think the individual elements may already have labels (at least the buttons do) - if not, it could also be good for them to always have the label (instead of it being conditional on the brightness).

Reply:
The attribute would just apply to the parent div i think